File: spring-boot-jersey/src/main/java/com/xphsc/jersey/service/impl/UserServiceImpl.java
package com.xphsc.jersey.service.impl;

import com.github.xphsc.bean.BeanByRefMapper;
import com.github.xphsc.collect.Lists;

import com.xphsc.jersey.model.User;
import com.xphsc.jersey.model.response.UserDTO;
import com.xphsc.jersey.repository.dao.UserRepository;
import com.xphsc.jersey.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

/**
 * Created by ${huipei.x} on 2017-2-25.
 */
@Service
public class UserServiceImpl implements UserService {

    @Autowired
    private UserRepository userRepository;

    @Override
    public List<UserDTO> listUser(){
        List<UserDTO> userDTOList= Lists.newArrayList();
        List<User>  userList=userRepository.findAll();
        userDTOList= BeanByRefMapper.copyByRefListMapper(userList,UserDTO.class);
        return userDTOList;
    }

    @Override
    public UserDTO  findUserByid(Integer id){

        User user= userRepository.findOne(id);
        UserDTO userDTO=BeanByRefMapper.copyByRefMapper(user,UserDTO.class);
        return userDTO;
    }

    @Override
    public UserDTO addUser(UserDTO userDTO) {
        User user=BeanByRefMapper.copyByRefMapper(userDTO,User.class);
        User userOne=userRepository.save(user);
        UserDTO userDTOOne=BeanByRefMapper.copyByRefMapper(userOne,UserDTO.class);
      return userDTOOne;
    }
}

File: spring-boot-kotlin/src/main/java/com/xphsc/kotlin/service/impl/UserServiceImpl.kt
package com.xphsc.kotlin.service.impl

import com.github.xphsc.bean.BeanByRefMapper
import com.github.xphsc.collect.Lists
import com.xphsc.kotlin.model.User
import com.xphsc.kotlin.model.response.UserDTO
import com.xphsc.kotlin.repository.dao.UserDao
import com.xphsc.kotlin.service.UserService
import org.springframework.stereotype.Service

/**
 * Created by ${huipei.x} on 2017-2-25
 */
@Service
class UserServiceImpl(private val userDao: UserDao) : UserService {

    override fun listUser(): List<UserDTO> {
        var userDTOList: List<UserDTO> = Lists.newArrayList<UserDTO>();
        var userList:List<User> = userDao.findAll();
        userDTOList = BeanByRefMapper.copyByRefListMapper(userList, UserDTO::class.java)
        return userDTOList;
    }

}